pkg/component/types.go

```diff
@@ -49,7 +49,7 @@ type SecretMount struct {
 // ComponentStatus is Status of components
 type ComponentStatus struct {
 	Context        string        `json:"context,omitempty"`
-	State          State         `json:"state"`
```

Comment:
We aren't, it's been replaced by using `string` instead of `State` which is easier to manipulate / if statements. All other places in the code such as `Storage` all use string. `component/types.go` is the only one that doesnt. See: https://github.com/redhat-developer/odo/blob/main/pkg/storage/types.go#L19